src/main/java/adris/altoclef/commands/EquipCommand.java

```diff
@@ -1,18 +1,41 @@
 package adris.altoclef.commands;
 
 import adris.altoclef.AltoClef;
+import adris.altoclef.Debug;
 import adris.altoclef.commandsystem.*;
 import adris.altoclef.tasks.misc.EquipArmorTask;
 import adris.altoclef.util.ItemTarget;
+import net.minecraft.item.Items;
 
 public class EquipCommand extends Command {
     public EquipCommand() throws CommandException {
-        super("equip", "Equips armor", new Arg(ItemList.class, "[armors]"));
+        super("equip", "Equips armor", new Arg(ItemList.class,
+                "[armors]"));
     }
 
     @Override
     protected void call(AltoClef mod, ArgParser parser) throws CommandException {
         ItemTarget[] items = parser.get(ItemList.class).items;
-        mod.runUserTask(new EquipArmorTask(items), this::finish);
+        if(items[0].matches(Items.DIAMOND)){
+            mod.runUserTask(new EquipArmorTask(Items.DIAMOND_CHESTPLATE, Items.DIAMOND_LEGGINGS, Items.DIAMOND_HELMET, Items.DIAMOND_BOOTS),this::finish);
+        }else if(items[0].matches(Items.NETHERITE_INGOT)){
```

Comment:
If anyone can think of a better way to do this, lmk. Since we are getting the item target, the material the armor is made from makes the most sense to use

Reply:
hit resolve conversation before merging

Reply:
Here's what you can do, it's not perfect but it might be useful.
```java
try {
    ItemTarget[] items = parser.get(ItemList.class).items;
} catch (CommandException e) {
    // We failed parsing our item list, but don't exit out yet: We might have custom keywords (ex. "iron"/"netherite")
    String[] args = parser.getArgs(); // Get the raw arguments as strings
    if (args.length != 1) {
        throw new CommandException("Invalid args, either item list or armor type (ex. \"iron\")"); // <-- Make this slightly better lol
    }
    String possibleMaterial = args[0];
    // TODO: Check if `possibleMaterial` is "netherite"/"iron"/"diamond" etc. If it's not any, throw a CommandExcepton
}
if (items.length == 0) {
    throw new CommandException("Please provide armor list or armor type");
}
if (items.length == 1) {
    // We might have parsed a single item, try that first.
    ItemTarget singleMaterialPossibly = items[0];
    if (singleMaterialPossibly.matches(Items.DIAMOND)) {
        // TODO: material "diamond"
        return;
    } else if (singleMaterialPossibly.matches(Items.LEATHER) {
        // TODO: material "leather"
        return;
    }
    // That's all of the material "gotcha's". The rest is just regular armor
}

// Check for bad items
for (ItemTarget target : items) {
    if (ALL_ARMORS.stream().noneMatch(armor -> target.matches(armor)))
        throw new CommandException("Not a valid armor item to equip: " + target);
}
mod.runUserTask(new EquipArmorTask(items), this::finish);
````

Reply:
I'd also make a method `getArmorOfMaterial(string)` and maybe a hashmap so you can reuse this function.
```java
private static final HashMap<String, Item[]> ARMORS_OF_SET = new HashMap<>() {
    put("diamond", ItemHelper.DIAMOND_ARMORS);
    ...
}
private static final Item[] getArmorOfMaterial(String material) {
    return ARMORS_OF_SET.get(material);
}
// Also might be helpful for the stuff above. I think you can use `Stream.combine` or something, there should be a 
private static final Item[] ALL_ARMORS = Stream.of(ItemHelper.DIAMOND_ARMORS, ItemHelper.IRON_ARMORS, ItemHelper.GOLD_ARMORS, etc...).flatmap(s -> s).toArray(Item[]::new);
```

Reply:
I get the first part with searching for words, but wouldn't it be simpler to simply parse the item as a string and look for diamond, leather, netherite, or iron? I don't see how 
`if (singleMaterialPossibly.matches(Items.DIAMOND)) {
        // TODO: material "diamond"
        return;
    } else if (singleMaterialPossibly.matches(Items.LEATHER) {
        // TODO: material "leather"
        return;
    }
    // That's all of the material "gotcha's". The rest is just regular armor`
is different from what is already being done.

I am also a little confused as to the helper functions. Can you explain a little more about what you mean?

Reply:
In your implementation, `@equip iron` doesn't work, you gotta do `@equip iron_ingot` which might feel weird

Reply:
agree with adri here

Reply:
Yea, I'll make these changes later